### PR TITLE
Add Subtraction and Negation to MPCTensor 

### PR DESCRIPTION
**Pull request by iamtrask**
https://github.com/OpenMined/PySyft/issues/1513

### CODE REVIEW

File: syft/core/frameworks/torch/tensor.py [hunk 1]
```diff
@@ -937,7 +937,7 @@ def share(self, bob, alice):
         x_alice.send(alice)
         x_bob.send(bob)
         x_pointer_tensor_dict = {alice: x_alice.child, bob: x_bob.child}
-        x_gp = _GeneralizedPointerTensor(x_pointer_tensor_dict).on(self)
+        x_gp = _GeneralizedPointerTensor(x_pointer_tensor_dict).on(self+0)
```
Comment (LaRiffle) [inline syft/core/frameworks/torch/tensor.py]: why are we doing this?

File: syft/core/frameworks/torch/tensor.py [hunk 2]
```diff
@@ -851,6 +851,16 @@ def __add__(self, other):
...
         return response
 
+    def __sub__(self, other):
+        gp_response = spdz.spdz_add(self.shares, spdz.spdz_neg(other.shares))
+        response = _MPCTensor(gp_response).wrap(True)
+        return response
+
+    def __neg__(self):
+        gp_response = spdz.spdz_neg(self.shares)
+        response = _MPCTensor(gp_response).wrap(True)
```
Comment (LaRiffle) [inline syft/core/frameworks/torch/tensor.py]: Does
2 -> mpc_x
mpc_x.__neg__()
returns -2?

File: test/torch_test.py [hunk 3]
```diff
@@ -985,6 +985,30 @@ def test_mpc_mul(self):
         self.mpc_mul(3, 5)
         self.mpc_mul(2 ** 12, 2 ** 12)
 
+    def test_mpc_matmul(self):
+        x = torch.LongTensor([[1, 2], [3, 4]])
+        y = torch.LongTensor([[5, 6], [7, 8]])
+
```
Comment (LaRiffle) [inline test/torch_test.py]: Maybe alos add a test where the result contains poisitve and negative values

Comment (iamtrask) [reply test/torch_test.py]: Good call - adding now

File: test/torch_test.py [hunk 4]
```diff
@@ -985,6 +985,30 @@ def test_mpc_mul(self):
...
+    def test_mpc_negation_and_subtraction(self):
+
+        x = torch.LongTensor([[1, 2], [3, 4]])
+        y = torch.LongTensor([[5, 6], [7, 8]])
+
+        x = x.share(bob, alice)
+        y = y.share(bob, alice)
+
+        z = -x
```
Comment (LaRiffle) [inline test/torch_test.py]: Same here, I would mix signs if possible

File: test/torch_test.py (hunk 3 above)
Comment (iamtrask) [reply test/torch_test.py]: @LaRiffle - added - passes on my machine

File: test/torch_test.py [hunk 5]
```diff
@@ -985,6 +985,30 @@ def test_mpc_mul(self):
...
         self.mpc_mul(2 ** 12, 2 ** 12)
 
+    def test_mpc_matmul(self):
+        x = torch.LongTensor([[1, 2], [3, 4]])
+        y = torch.LongTensor([[5, 6], [7, 8]])
+
+        x = x.share(bob, alice)
+        y = y.share(bob, alice)
+
+        assert (x.mm(y).get() - torch.LongTensor([[18, 22], [43, 49]])).abs().sum() < 5
```
[the remark this answers is not in the document]
Comment (iamtrask) [reply test/torch_test.py]: no because there's numerical instability we need to sort out which is outside of the scope of this ticket

File: test/torch_test.py (hunk 4 above)
Comment (iamtrask) [reply test/torch_test.py]: cool - fixing.. one sec.

Comment (iamtrask) [reply test/torch_test.py]: crap... good call... this part is broken

Comment (iamtrask) [reply test/torch_test.py]: if they're all negative or all positive it works great, but not if it's a mix

File: test/torch_test.py (hunk 3 above)
Comment (LaRiffle) [reply test/torch_test.py]: 👍

File: test/torch_test.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (LaRiffle) [reply test/torch_test.py]: Isn't it solved? I thought the PR I made solved it :/

[the remark this answers is not in the document]
Comment (iamtrask) [reply test/torch_test.py]: oh... maybe i don't have that...

[the remark this answers is not in the document]
Comment (LaRiffle) [reply test/torch_test.py]: https://github.com/OpenMined/PySyft/pull/1512
